runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformResult.java

```diff
@@ -33,7 +33,7 @@
 /**
  * The result of evaluating an {@link AppliedPTransform} with a {@link TransformEvaluator}.
  */
-public interface TransformResult {
+public interface TransformResult<InputT, OutputT> {
```

Comment:
I don't like this `OutputT` because of `ParDo.BoundMulti`

I'm happy with the `InputT` however - especially given that you're using rawtypes in the `ParDoEvaluator`